packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -247,48 +253,64 @@ export class RestAPIClient {
 	}
 
 	/**
-	 * Deletes a site.
+	 * Deletes a site belonging to the user.
+	 *
+	 * This method will perform a quick check to ensure the target site
+	 * belongs to the user.
 	 *
-	 * If the target site has an upgrade purchased within the
+	 * Additionally, if the target site has an upgrade purchased within the
 	 * sandboxed environment, it can be deleted without first
 	 * cancelling the subscription.
 	 *
 	 * Otherwise the active subscription must be first cancelled
-	 * otherwise the REST API will throw a HTTP 403 status.
+	 * or else the REST API will throw a HTTP 403 status.
 	 *
 	 * @param {SiteDetails} expectedSiteDetails Expected details for the site to be deleted.
 	 * @returns {SiteDeletionResponse | null} Null if deletion was unsuccessful or not performed. SiteDeletionResponse otherwise.
 	 */
-	async deleteSite( expectedSiteDetails: SiteDetails ): Promise< SiteDeletionResponse | null > {
-		if ( ! expectedSiteDetails.url.includes( 'e2e' ) ) {
-			console.warn( `Aborting site deletion: target is not a test site.` );
+	async deleteSite( targetSite: {
+		id: number;
+		domain: string;
+	} ): Promise< SiteDeletionResponse | null > {
+		// Every new testing site since 2021 has the prefix `e2e` of some sort.
+		if ( ! targetSite.domain.includes( 'e2e' ) ) {
 			return null;
 		}
 
-		const mySites = await this.getAllSites();
-		const myAccountInformation = await this.getMyAccountInformation();
+		const targetDomain = targetSite.domain.startsWith( 'http' )
+			? targetSite.domain.replace( 'https://', '' )
+			: targetSite.domain;
```

Comment:
This function strips the `https://` prefix from the site to create a slug.

This is because the `/all-domains/` endpoint only returns the site slug, while other endpoints that we normally work with `eg. /me/sites` return the fully qualified URL.